mne/io/tests/test_meas_info.py

```diff
@@ -339,6 +339,15 @@ def test_check_consistency():
     info2['chs'][2]['ch_name'] = 'b'
     assert_raises(RuntimeError, info2._check_consistency)
 
+    # Duplicates appended with running numbers
+    info2['ch_names'][2] = 'b'
+    with warnings.catch_warnings(record=True) as w:
+        info2._check_consistency()
+    assert_equal(len(w), 1)
+    assert_true(all(ww.message.message.startswith('Channel names are not '
+                                                  'unique') for ww in w))
+    assert_true(info2['ch_names'][2] == 'b-1')
```

Comment:
can you test that you have now b-0 and b-1?

can you check also that it works with 2 duplicates channels?